pkg/oci/utils.go

```diff
@@ -161,57 +196,115 @@ func networkConfig(ocispec CompatOCISpec) (vc.NetworkConfig, error) {
 	return netConf, nil
 }
 
-// PodConfig converts an OCI compatible runtime configuration file
-// to a virtcontainers pod configuration structure.
-func PodConfig(runtime RuntimeConfig, bundlePath, cid, console string) (*vc.PodConfig, *CompatOCISpec, error) {
-	configPath := filepath.Join(bundlePath, "config.json")
+// getConfigPath returns the full config path from the bundle
+// path provided.
+func getConfigPath(bundlePath string) string {
+	return filepath.Join(bundlePath, "config.json")
+}
+
+// ParseConfigJSON unmarshals the config.json file.
+func ParseConfigJSON(bundlePath string) (CompatOCISpec, error) {
+	configPath := getConfigPath(bundlePath)
 	ociLog.Debugf("converting %s", configPath)
 
 	configByte, err := ioutil.ReadFile(configPath)
 	if err != nil {
-		return nil, nil, err
+		return CompatOCISpec{}, err
 	}
 
 	var ocispec CompatOCISpec
-	if err = json.Unmarshal(configByte, &ocispec); err != nil {
-		return nil, nil, err
+	if err := json.Unmarshal(configByte, &ocispec); err != nil {
+		return CompatOCISpec{}, err
 	}
 
-	rootfs := ocispec.Root.Path
-	if !filepath.IsAbs(rootfs) {
-		rootfs = filepath.Join(bundlePath, ocispec.Root.Path)
+	return ocispec, nil
+}
+
+// GetContainerInfo returns the container status and its pod ID.
+func GetContainerInfo(containerID string) (vc.ContainerStatus, string, error) {
+	podStatusList, err := vc.ListPod()
+	if err != nil {
+		return vc.ContainerStatus{}, "", err
 	}
-	ociLog.Debugf("container rootfs: %s", rootfs)
 
-	cmd := vc.Cmd{
-		Args:         ocispec.Process.Args,
-		Envs:         cmdEnvs(ocispec, []vc.EnvVar{}),
-		WorkDir:      ocispec.Process.Cwd,
-		User:         strconv.FormatUint(uint64(ocispec.Process.User.UID), 10),
-		PrimaryGroup: strconv.FormatUint(uint64(ocispec.Process.User.GID), 10),
-		Interactive:  ocispec.Process.Terminal,
-		Console:      console,
+	for _, podStatus := range podStatusList {
+		for _, containerStatus := range podStatus.ContainersStatus {
+			if containerStatus.ID == containerID {
+				return containerStatus, podStatus.ID, nil
+			}
+		}
 	}
 
-	cmd.SupplementaryGroups = []string{}
-	for _, gid := range ocispec.Process.User.AdditionalGids {
-		cmd.SupplementaryGroups = append(cmd.SupplementaryGroups, strconv.FormatUint(uint64(gid), 10))
+	return vc.ContainerStatus{}, "", fmt.Errorf("Could not find any pod/container ID %s", containerID)
+}
+
+// GetContainerType determines which type of container matches the annotations
+// table provided.
+func GetContainerType(annotations map[string]string) (ContainerType, error) {
```

Comment:
There is a main issue about doing this kind of thing, that is we will end up looping over all pods and containers while we already have the annotations (or the ociSpec handler) with us. I mean from a runtime perspective, this will lead us to loop many times over the list of pods and containers while we already have the information in one of our structure.